packages/manager/modules/pci/src/components/project/storages/databases/user.class.js

```diff
@@ -0,0 +1,59 @@
+import Base from './base.class';
+
+export default class User extends Base {
+  constructor({
+    createdAt,
+    id,
+    roles,
+    status,
+    username,
+    acls,
+    categories,
+    channels,
+    commands,
+    keys,
+  }) {
+    super();
+    this.updateData({
+      createdAt,
+      id,
+      rolesArray: roles,
+      status,
+      username,
+      acls,
+      categories,
+      channels,
+      commands,
+      keys,
+    });
+  }
+
+  updateData({
+    createdAt,
+    id,
+    rolesArray,
+    status,
+    username,
+    acls,
+    categories,
+    channels,
+    commands,
+    keys,
+  }) {
+    Object.assign(this, {
+      createdAt,
+      id,
+      rolesArray,
+      status,
+      username,
+      acls,
+      categories,
+      channels,
+      commands,
+      keys,
+    });
+    this.roles = this.rolesArray?.map((role, index) => {
+      return { id: index, name: role };
+    });
```

Comment:
it's better to have an empty array instead of `undefined`.

```suggestion
    this.roles = (this.rolesArray || []).map((role, index) => {
      return { id: index, name: role };
    });
```

Reply:
Hi @mohammed-zahaf , here we really need to have the roles set to undefined in some cases, as there are 3 possibilities depending on the engine:
- the user have one or many roles
- the user does not have any role, but could have
- the user does not have any role because the engine does not allow roles on user

We use the users page component which is global to all pci universe, where a check is made on  the users role property to show or hide the "role" column. If it is an empty array, the column will be shown,  which we do not want if the engine does not allow users.

Reply:
Ok got it.